src/activity_handlers.cpp

```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
         return;
     }
 
+    bool has_table_nearby = false;
+    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 1 ) ) {
+        if( g->m.furn( pt ) == furn_id( "f_table" ) ) {
+            has_table_nearby = true;
+        }
+    }
+    // workshop butchery (full) prequisites
+    if( action == BUTCHER_FULL ) {
+            if( g->m.furn( u.pos() ) != furn_id( "f_butcher_rack" ) ){
+                u.add_msg_if_player( m_info, _( "You need a butchering rack to perform a full butchery." ) );
+                act.set_to_null();
+                return;
+            }
+            if ( !has_table_nearby ) {
+                u.add_msg_if_player( m_info, _( "You need a table nearby to perform a full butchery." ) );
+                act.set_to_null();
+                return;
+            }
+            if( ( !u.has_quality( quality_id( "CUT" ) ) || !u.has_quality( quality_id( "CUT_FINE" ) ) ) &&
+                ( !u.has_quality( quality_id( "SAW_W" ) ) || !u.has_quality( quality_id( "SAW_M" ) ) ) ) {
+                u.add_msg_if_player( m_info, _( "You need tools that can cut and saw to perform a full butchery." ) );
+                act.set_to_null();
+                return; 
+            }
+    }
+
+    item corpse_item = items[act.index];
     const mtype *corpse = items[act.index].get_mtype();
+
+    if( action == F_DRESS && ( corpse_item.has_flag( "FIELD_DRESS" ) || corpse_item.has_flag( "FIELD_DRESS_FAILED" ) ) ) {
+            u.add_msg_if_player( m_info, _( "This corpse is already field dressed." ) );
+            act.set_to_null();
+            return;
+    }
+    // applies to all butchery actions
+    if( corpse->in_species( HUMAN ) && ( !u.has_trait_flag( "CANNIBAL" ) || !u.has_trait_flag( "PSYCHOPATH" ) || !u.has_trait_flag( "SAPIOVORE" ) ) ) {
+        add_msg( m_info, _( "Why would you do this to mortal remains of a fellow human?" ) );
```

Comment:
Personal thought: This should still be allowable, but give a morale penalty if you're still bent on preventing it. Historically this kind of thing has been skewed towards "allow the player to do detrimental things if they really want to", like how you can consume human meat (even if it gives a huge morale penalty) and the like.